bindings/python/triplestore/README.md

```diff
@@ -0,0 +1,117 @@
+Triplestore
+===========
+> A Python package encapsulating different triplestores using the strategy
+> design pattern.
+
+This package has no dependencies outside the standard library, but the
+triplestore backends have.
+
+The main class is Triplestore, who's `__init__()` method takes the name of the
+backend to encapsulate as first argument.  It's interface is strongly inspired
+by rdflib.Graph, but simplified when possible to make it easy to use.  Some
+important differences:
+- all IRIs are represented by Python strings
+- blank nodes are strings starting with "_:"
+- literals are constructed with Literal()
+
+The module already provides a set of pre-defined namespaces that simplifies
+writing IRIs. For example:
+
+```python
+>>> from triplestore import RDFS, OWL
+>>> RDFS.subClassOf
+'http://www.w3.org/2000/01/rdf-schema#subClassOf'
+```
+
+New namespaces can be created using the Namespace class, but are usually
+added with the bind() method:
+
+```python
+>>> from triplestore import Triplestore
+>>> ts = Triplestore(backend="rdflib")
+>>> ONTO = ts.bind("onto", "http://example.com/onto#")
+>>> ONTO.MyConcept
+'http://example.com/onto#MyConcept'
+```
+
+New triples can added either with the parse() method (for backends that support
+it) or the add() and add_triples() methods.
+
+```python
+# en(msg) is a convinient function for adding english literals.
+# It is equivalent to ``triplestore.Literal(msg, lang="en")``.
+>>> from triplestore import en
+>>> ts.parse("onto.ttl", format="turtle")
+>>> ts.add_triples([
+...     (ONTO.MyConcept, RDFS.subClassOf, OWL.Thing),
+...     (ONTO.MyConcept, RDFS.label, en("My briliant ontological concept.")),
+... ])
+```
+
+For backends that support it can the triplestore be serialised using
+serialize():
+
+```python
+>>> ts.serialize("onto2.ttl")
+```
+
+A set of convenient functions exists for simple queries, including
+triples(), subjects(), predicates(), objects(), subject_predicates(),
+subject_objects(), predicate_objects() and value().  Except for value(),
+they return the result as generators. For example:
+
+```python
+>>> list(ts.objects(subject=ONTO.MyConcept, predicate=RDFS.subClassOf))
+['http://www.w3.org/2002/07/owl#Thing']
+```
+
+The query() and update() methods can be used to query and update the
+triplestore using SPARQL.
+
+Finally Triplestore has two specialised methods add_mapsTo() and
+add_function() that simplify working with mappings.  add_mapsTo() is
+convinient for defining new mappings:
+
+```python
+>>> from triplestore import Namespace
+>>> META = Namespace("http://onto-ns.com/meta/0.1/MyEntity#")
+>>> ts.add_mapsTo(ONTO.MyConcept, META.my_property)
+```
+
+It can also be used with DLite and SOFT7 data models.  Here we repeat
+the above with DLite:
+
+```python
+>>> import dlite
+>>> meta = dlite.get_entity("http://onto-ns.com/meta/0.1/MyEntity")
+>>> ts.add_mapsTo(ONTO.MyConcept, meta, "my_property")
```

Comment:
```suggestion
import dlite
meta = dlite.get_instance("http://onto-ns.com/meta/0.1/MyEntity")
ts.add_mapsTo(ONTO.MyConcept, meta, "my_property")
```

Reply:
Also we should have an example that works out of the box. Is MyEntity there as example in storages?